diff -ur openssl-1.1.1a.orig/crypto/threads_win.c openssl-1.1.1a/crypto/threads_win.c
--- openssl-1.1.1a.orig/crypto/threads_win.c	2018-11-20 14:35:38.000000000 +0100
+++ openssl-1.1.1a/crypto/threads_win.c	2018-12-11 01:14:42.469846843 +0100
@@ -25,7 +25,8 @@
     }
 
     /* 0x400 is the spin count value suggested in the documentation */
-    if (!InitializeCriticalSectionAndSpinCount(lock, 0x400)) {
+    /* on Win9x InitializeCriticalSectionAndSpinCount calls InitializeCriticalSection and returns 0 */
+    if (!InitializeCriticalSectionAndSpinCount(lock, 0x400) && (GetLastError() != ERROR_SUCCESS)) {
         OPENSSL_free(lock);
         return NULL;
     }
