diff -ur openssl-1.1.1b.orig/e_os.h openssl-1.1.1b/e_os.h
--- openssl-1.1.1b.orig/e_os.h	2019-02-26 15:15:30.000000000 +0100
+++ openssl-1.1.1b/e_os.h	2019-03-31 18:36:38.963598608 +0200
@@ -128,7 +128,7 @@
         * at run-time [DSO_global_lookup is recommended]...
         */
 #    include <winsock2.h>
-#    include <ws2tcpip.h>
+#    include <wspiapi.h>
        /* yes, they have to be #included prior to <windows.h> */
 #   endif
 #   include <windows.h>
