diff -ur openssl-1.1.1c.orig/crypto/threads_win.c openssl-1.1.1c/crypto/threads_win.c
--- openssl-1.1.1c.orig/crypto/threads_win.c	2019-05-28 15:12:21.000000000 +0200
+++ openssl-1.1.1c/crypto/threads_win.c	2019-08-07 23:49:12.361673061 +0200
@@ -27,8 +27,21 @@
 #if !defined(_WIN32_WCE)
     /* 0x400 is the spin count value suggested in the documentation */
     if (!InitializeCriticalSectionAndSpinCount(lock, 0x400)) {
-        OPENSSL_free(lock);
-        return NULL;
+        /* check if win9x */
+        OSVERSIONINFOA osv;
+
+        osv.dwOSVersionInfoSize = sizeof(OSVERSIONINFOA);
+        osv.dwPlatformId = 0x0010400;
+
+        GetVersionExW((LPOSVERSIONINFOA) &osv);
+
+        if (osv.dwPlatformId & VER_PLATFORM_WIN32_NT) {
+            OPENSSL_free(lock);
+            return NULL;
+        }
+
+        /* let's hope it works */
+        InitializeCriticalSection(lock);
     }
 #else
     InitializeCriticalSection(lock);
